R/mail.R

```diff
@@ -6,110 +6,249 @@
+
+  # Table rows
+  rows <- XML::getNodeSet(doc_obj, "//tr")
+
+  # Skip header row
+  data_rows <- rows[-1]
+
+  # Vector for link storage
+  links <- c()
+
+  # Extract the date and link from each row
+  for (row in data_rows) {
+    # Date in YYYYMM format
+    date_extracted <- XML::xpathSApply(row, ".//td[1]", XML::xmlValue)
+    date_cleaned <- stringi::stri_replace_last_regex(date_extracted, pattern = ":$", replacement = "")
+    date_cleaned <- stringi::stri_trim_both(date_cleaned)
+    # Parse the date
+    date_parsed <- as.Date(paste0("01 ", date_cleaned), format = "%d %B %Y")
+    if (is.na(date_parsed)) {
+      warning("Date could not be parsed: ", date_cleaned)
+      next
+    }
+    year_month <- format(date_parsed, "%Y%m")
+
+    # Check if date is within range
+    if (year_month >= start_year_month & year_month <= end_year_month) {
+      # Get href from column 3
+      link_nodes <- XML::xpathSApply(row, ".//td[3]/a", XML::xmlGetAttr, 'href')
+      if (length(link_nodes) == 0) {
+        warning("No link found in row for date: ", date_cleaned)
+        next
+      }
+      # Store the link in links
+      link <- link_nodes[1]
+      links <- c(links, link)
     }
   }
 
-  destination <- vector()
-  # File downloading loop
-  for (i in files){
+  # Vector for downloaded files
+  downloaded_files <- c()
+  for (i in seq_along(links)) {
+    link <- links[i]
 
-    #split filename from url and create download destination out of it
-    splits <- stringi::stri_split_fixed(i, "/")
-    destination[[i]] <- paste0(splits[[1]][[length(splits[[1]])]])
+    # Extract the name without the .txt.gz extension
+    base_name <- gsub("\\.txt\\.gz$", "", link)
 
-    #download file and place it at the destination
-    httr::GET(i, httr::write_disk(destination[[i]], overwrite=TRUE))
-  }
+    # Parse the date from the base name
+    date_parsed <- as.Date(paste0("01-", base_name), format = "%d-%Y-%B")
+    if (is.na(date_parsed)) {
+      warning("Could not parse date from link: ", link)
+      next
+    }
+    year_month_clean <- format(date_parsed, "%Y%m")
+
+    # Download URL
+    txt_url <- paste0(mailing_list, gsub("\\.gz$", "", link))
+    gz_url <- paste0(mailing_list, link)
+
+    # Attempt to download the .txt file first
+    download_url <- txt_url
+    response <- httr::GET(download_url, httr::timeout(60))
+
+    if (httr::status_code(response) != 200) {
+      # Fallback to .gz file if .txt is unavailable
+      download_url <- gz_url
+      response <- httr::GET(download_url, httr::timeout(60))
+      if (httr::status_code(response) != 200) {
+        cat("Both .txt and .gz downloads failed for link: ", link, "\n")
+        next
+      }
+    }
 
-  #Return filenames
-  return(destination)
+    # Define the destination file
+    dest <- file.path(save_folder_path, paste0('kaiaulu_', year_month_clean, '.mbox'))
+
+    # Print diagnostic info
+    cat("Downloading: ", download_url, "\n")
+    cat("Saving to: ", dest, "\n")
```

Comment:
Add a verbose parameter to the function, and specif these under "if verbose == true". Any prints should be optional.